contracts/Vault2.sol

```diff
@@ -1,6 +1,20 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault2 {
-    
-}
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+contract Vault2 is ERC20 {
+    constructor() ERC20('VaultToken', "VAULT") {
+    }
+
+    function mint() external payable {
+        _mint(msg.sender, msg.value);
+    }
+
+    function burn(uint _amount) external {
+        require(balanceOf(msg.sender) >= _amount, "Insufficient balance");
+        bool _success = payable(msg.sender).send(_amount);
```

Comment:
Not recommended to use send. Best to use `call` here. Using send needs an extra check since it returns a boolean and doesn't revert automatically